src/byte_stream.rs

```diff
@@ -0,0 +1,579 @@
+use crate::html5::tokenizer::{CHAR_CR, CHAR_LF};
+use std::io::Read;
+use std::{fmt, io};
+
+/// Encoding defines the way the buffer stream is read, as what defines a "character".
```

Comment:
The bytestream is a slightly improved version of the regular "bytes.rs" iterator.